docs/content/guides/dagster/enriching-with-software-defined-assets.mdx

```diff
@@ -0,0 +1,407 @@
+description: Understand when, why, and how to use software-defined assets (SDAs) in Dagster, as well as how SDAs work with other core Dagster concepts.
+---
+
+# Upgrading To Software-Defined Assets
+
+If you want to understand when, why, and how to use software-defined assets in Dagster, this guide is for you. We'll also demonstrate what some common Dagster jobs look like before and after using software-defined assets.
+
+Before we jump in, here's a quick refresher:
+
+- An **asset** is a persistent object in storage, such as a table, machine learning (ML) model, or file.
+- An [**op**](/concepts/ops-jobs-graphs/ops) is the core unit of computation in Dagster. For example, an op might accept tabular data as input and produce transformed tabular data as output.
+- A [**graph**](/concepts/ops-jobs-graphs/jobs-graphs) is a directed acyclic graph of ops or other graphs, which execute in order and pass data to each other.
+- A [**software-defined asset**](/concepts/assets/software-defined-assets) is a declaration of an asset that should exist and a description of how to compute it: the op or graph that needs to run and the upstream assets that it should run on.
+
+**Software-defined assets aren't a replacement for Dagster's core computational concepts**. Think of them as a top layer that links ops, graphs, and jobs to the long-lived objects they interact with.
+
+---
+
+## Why use software-defined assets?
+
+Using software-defined assets means building Dagster jobs in a way that declares _ahead of time_ the assets they produce and consume. This is different than using the `AssetMaterialization` API, which only informs Dagster at runtime about the assets a job interacted with.
+
+Preemptively declaring assets offers distinct advantages, including:
+
+### Lineage
+
+As software-defined assets know what other assets they depend on, an asset's lineage can be [viewed easily in Dagit](/concepts/assets/software-defined-assets#viewing-assets-in-dagit).
+
+Assets help track and define cross-job dependencies. For example, when viewing a job that materializes assets, you can navigate to the jobs that produce the assets that it depends on. Additionally, when an upstream asset has been updated more recently than a downstream asset, Dagster will indicate that the downstream asset might be out of date.
+
+### Direct operation
+
+Using software-defined assets enables you to directly operate them in Dagit. On the [Assets page](/concepts/dagit/dagit#assets), you can:
+
+- View the materialization history of an asset
+- Check when the next materialization will occur
+- Launch a run to materialize the asset
+
+Drilling down to the [Asset Details page](/concepts/dagit/dagit#asset-details) for an individual asset shows the sensors or schedules for any jobs targeting the asset. It also includes a **(Re)Materialize** button, allowing you to launch a run that rematerializes the asset, including its ancestors or descendants.
+
+### Improved code ergonomics
```

Comment:
The way I was envisioning the structure of this section was that there are two separate top-level reasons that SDAs have "Improved Code Ergonomics". One is that using software-defined assets often means writing less code. The other is that you get to avoid a tradeoff between easy dependency tracking and manageable organization.

Not sure if my version did a good job, but I think the current structure mixes these in a kind of confusing way.

Reply:
Gotcha! I admit I didn't catch that in the original version, but I see it now that you've clarified. I'll work some magic.